src/Geometry.cpp

```diff
@@ -528,48 +519,49 @@ int Geometry::findFSRId(LocalCoords* coords) {
   std::size_t fsr_key_hash = key_hash_function(getFSRKey(coords));
 
   /* If FSR has not been encountered, update FSR maps and vectors */
-  if (_FSR_keys_map.find(fsr_key_hash) == _FSR_keys_map.end()){
-
-    /* Get the cell that contains coords */
-    Cell* cell = findCellContainingCoords(curr);
-    
-    /* Get the lock */
-    omp_set_lock(_num_FSRs_lock);
-
-    /* Recheck to see if FSR has been added to maps after getting the lock */
-    if (_FSR_keys_map.find(fsr_key_hash) != _FSR_keys_map.end())
-      fsr_id = _FSR_keys_map.at(fsr_key_hash)._fsr_id;
+  if (!_FSR_keys_map.contains(fsr_key_hash)){
+
+    /* Try to get a clean copy of the fsr_id, adding the FSR data 
+       if necessary where -1 indicates the key was already added */
+    fsr_id = _FSR_keys_map.insert_and_get_count(fsr_key_hash, NULL);
+    if( fsr_id == -1)
+    {
+      fsr_data volatile* fsr;
+      do{
```

Comment:
Agreed, I *think* this overwhelmingly the case. And if not, it should be in
the future since it makes for cleaner code.
On Jun 30, 2015 10:18 AM, "Nathan A. Gibson" <notifications@github.com>
wrote:

> In src/Geometry.cpp
> <https://github.com/mit-crpg/OpenMOC/pull/157#discussion_r33598424>:
>
> > -
> > -    /* Get the lock */
> > -    omp_set_lock(_num_FSRs_lock);
> > -
> > -    /* Recheck to see if FSR has been added to maps after getting the lock */
> > -    if (_FSR_keys_map.find(fsr_key_hash) != _FSR_keys_map.end())
> > -      fsr_id = _FSR_keys_map.at(fsr_key_hash)._fsr_id;
> > +  if (!_FSR_keys_map.contains(fsr_key_hash)){
> > +
> > +    /* Try to get a clean copy of the fsr_id, adding the FSR data
> > +       if necessary where -1 indicates the key was already added */
> > +    fsr_id = _FSR_keys_map.insert_and_get_count(fsr_key_hash, NULL);
> > +    if( fsr_id == -1)
> > +    {
> > +      fsr_data volatile* fsr;
> > +      do{
>
> I think we're putting spaces before { in the vast majority of cases.
> Where are you seeing us not doing it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/OpenMOC/pull/157/files#r33598424>.
>